### ISSUE TITLE
array-callback-return should report .flatMap()

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#12235]
[Array.prototype.flatMap](https://www.ecma-international.org/ecma-262/10.0/index.html#sec-array.prototype.flatmap) was added in ECMAScript 2019

**What rule do you want to change?**

`array-callback-return`

**Does this change cause the rule to produce more or fewer warnings?**

more

**How will the change be implemented? (New option, new default behavior, etc.)?**

new default behavior

**Please provide some example code that this change will affect:**

```js
/*eslint array-callback-return: "error"*/

foo.flatMap(x => {})
```

**What does the rule currently do for this code?**

No errors

**What will the rule do after it's changed?**

Error: `Expected to return a value in arrow function.` 

**Are you willing to submit a pull request to implement this change?**

Yes.

Can this be a semver-minor default behavior change? It looks too 'small' for an option.

**Comment by platinumazure** [comment]
I'm worried about the potential breaking change impact.

Could we look for ecmaVersion >= 2019 in a semver-minor update, then maybe remove that check in the next major release?

**Comment by mdjermanovic** [comment]
> Could we look for ecmaVersion >= 2019 in a semver-minor update, then maybe remove that check in the next major release?

I think it's a good idea.

Might be technically a bit more correct to check if an ES2019+ environment is activated rather than parser, but I don't think it's possible from within a rule at this moment.

**Comment by g-plane** [comment]
I don't think it's a breaking change, though it may break user's tests or CI.
